vllm/core/block/cpu_gpu_block_allocator.py

```diff
@@ -283,22 +304,31 @@ def mark_blocks_as_computed(self, block_ids: List[int]) -> None:
         device = Device.GPU
         return self._allocators[device].mark_blocks_as_computed(block_ids)
 
+    def get_computed_block_ids(self, prev_computed_block_ids: List[int],
+                               block_ids: List[int]) -> List[int]:
+        # Prefix caching only supported on GPU.
+        device = Device.GPU
+        return self._allocators[device].get_computed_block_ids(
+            prev_computed_block_ids, block_ids)
+
     def get_common_computed_block_ids(
-            self, seq_block_ids: List[List[int]]) -> List[int]:
+            self, computed_seq_block_ids: List[List[int]]) -> List[int]:
         # Prefix caching only supported on GPU.
         device = Device.GPU
         return self._allocators[device].get_common_computed_block_ids(
-            seq_block_ids)
+            computed_seq_block_ids)
 
     @property
     def all_block_ids(self) -> FrozenSet[int]:
         return frozenset(self._block_ids_to_allocator.keys())
 
     def promote_to_immutable_block(self, block: Block) -> BlockId:
-        raise NotImplementedError
+        device = Device.GPU
+        return self._allocators[device].promote_to_immutable_block(block)
```

Comment:
yeah

Reply:
add some comment when it's used? (I think they should be removed but seems I miss a case)

Reply:
You actually right, this is cpu-gpu allocator, so it is not doing the actual CoW or promo, since it is done only by the specific Naive or Prefix allocators, and they have these functions define via the base class BlockAllocator. Good catch!